Uplift/Models/OpenHours.swift

```diff
@@ -0,0 +1,72 @@
+//
+//  OpenHours.swift
+//  Uplift
+//
+//  Created by alden lamp on 9/19/23.
+//  Copyright © 2023 Cornell AppDev. All rights reserved.
+//
+
+import Foundation
+
+struct OpenHours {
+
+    var rawHours: [Int: [QLOpenHours]]
+
+    var todaysHours: [QLOpenHours]? {
+        rawHours[Date().getIntegerDayOfWeekToday()]
+    }
+
+    var tomorrowsHours: [QLOpenHours]? {
+        rawHours[Date().getIntegerDayOfWeekTomorrow()]
+    }
+
+    init(openHours: [QLOpenHours]) {
+        rawHours = [:]
+        for openHour in openHours {
+            if rawHours.keys.contains(openHour.day) {
+                rawHours[openHour.day]?.append(openHour)
+            } else {
+                rawHours[openHour.day] = [openHour]
+            }
+        }
+    }
+
+    func isStatusChangingSoon(_ date: Date = Date()) -> Bool {
+        return todaysHours?.first { $0.willChangeSoon(date) == true } != nil
+    }
+
+    func isOpen(_ date: Date = Date()) -> Bool {
+        guard todaysHours != nil else {return false}
+        return !todaysHours!.filter { $0.isDateInRange(date) }.isEmpty
+    }
+
+    func getHoursString(_ date: Date = Date()) -> String {
+        let strFormat: String
+
+        // Gym is open
+        if isOpen(date), let nowHourRange = todaysHours?.filter({  $0.isDateInRange(date) }).first {
+            strFormat = nowHourRange.endTime.getHourFormat()
+            let closeTime = nowHourRange.endTime.getStringOfDatetime(format: strFormat)
+            return ClientStrings.Home.gymDetailCellClosesAt + closeTime
+
+        } else {
```

Comment:
could we make this if else sequence cases instead, like case .isOpen or case .willOpenToday or case .willOpenTomorrow to make the code more readable?

Reply:
It makes more sense to have them as if let statements because we are checking for the existence of different properties/function values. I've made some updates to make it more readable/look more like switch.